Provisioning/Vault/audit.tf

```diff
@@ -1,7 +1,17 @@
-resource "vault_audit" "file" {
+resource "vault_audit" "stdout" {
```

Comment:
Why not file inside the container with some retention on it?

Reply:
Vault does not work if otlp is down w/o a second audit device

Reply:
Files in Kubernetes volumes cannot be easily rotated (Promtail could be used if the file were on the Kubernetes node). Some solutions are suggested here: https://support.hashicorp.com/hc/en-us/articles/13444259109523-Information-in-regards-to-Audit-Log-when-running-Vault-in-Kubernetes

I don't like the idea of installing another piece of code just for the purpose of rotating a specific file in a Pod of a specific application (with specific UID/GID), when actually have a completely working Kubernetes logging stack that takes care of rotating the logs for us.

The file inside the container is a legit approach though. One could configure an `emptyDir` as @pree suggested me this week.

```yaml
volumes: null
  - name: audit
    emptyDir: {}

volumeMounts: null
  - mountPath: /vault/audit-log
    name: audit
```

This only has one caveat that you need to be aware of. The volume is growing constantly and will only be purged [once the Pod is "removed form a node"](https://kubernetes.io/docs/concepts/storage/volumes/#emptydir). I think you might even experience a `CrashLoopBackOff` failure or similar if you simply let the audit log grow too large.. never tested it though. But you could test it out easily with that playground code and then let the volume grow 100% once to see how the application behaves. In any case, the logs of the volume would not be stored away for retention purposes. This might be what you want, might not, depends on the requirements for retention on this audit path (vault -> file on pvc).

For this PR here I still have the `stdout` approach for the "backup path" (the logging path that ensures Vault keeps running in case there are failures with the "primary" audit device and logging path, tpc in this example). This way, the logs are cleanly rotated and stored in the default Kubernetes logging stack. Downside here is that everyone can see the audit logs with kubectl logs.

For forwarding to Loki I decided to "filter out the audit logs" and [drop](https://grafana.com/docs/loki/latest/send-data/promtail/stages/drop) them so that you could decide to give access to "only the vault server" logs in Loki. My perspective is that anyone who can do `kubectl exec/logs` on the platform level can actually do pretty much anything and that logs are to be treated sensitive by default (not just Vault, any app). So if you want to give anyone access to your logs (e.g., business), you might do this through higher-level abstractions and tools/filtering like Loki. Might be more safe.

Reply:
Yes, but I think having a backup audit device in something like a `emptyDir` should only be used if uptime is more valued than audit-ability. otlp, fluentBit, etc could be configured HA, so that it's unlikely to fail that often / suddenly fails and could send the audit logs to two different endpoints (like loki & file) too.
This would also not expose audit logs anymore if someone would have `kubectl exec/logs` access to a vault pod. 

I think this really depends on the installation and environment where it's deployed. There won't be a go-to solution for everyone, as some wouldn't mind loosing some audit logs for uptime, others would rather have Vault stop working if audit logs cannot be written anymore.

I think the biggest value added by otlp, fluentBit, etc is, that it could cache audit logs in a persistent volume if the external target (like splunk or log analytics) is temporarily down, so that Vault wouldn't stop working if there is an outage.